src/main/java/graphql/nadel/engine/transformation/FieldTransformation.java

```diff
@@ -1,59 +1,19 @@
 package graphql.nadel.engine.transformation;
 
 import graphql.analysis.QueryVisitorFieldEnvironment;
-import graphql.execution.nextgen.result.NamedResultNode;
-import graphql.execution.nextgen.result.ObjectExecutionResultNode;
+import graphql.execution.MergedField;
+import graphql.execution.nextgen.result.ExecutionResultNode;
 import graphql.util.TraversalControl;
 
-import java.util.List;
-
 public interface FieldTransformation {
+
     TraversalControl apply(QueryVisitorFieldEnvironment environment);
 
-    /**
-     * The idea here is to convert back delegate result  for the field
-     * and convert it back to what it should be on merged schema side.
-     * Unapply should know which fields to fetch and return.
-     * E.g. (this is query mixed with pseudo nadel for brevity)
-     * Client query:
-     * <pre>
-     * query {
-     *     fieldA &lt;= $source.originalName {
-     *          alias: fieldB
-     *          hydration &lt;= uses fieldC as a parameter
-     *     }
-     * }
-     * Delegate query:
-     * query {
-     *     originalName {
-     *         alias: fieldB
-     *         fieldC
-     *     }
-     * }
-     *
-     * Delegate result:
-     * originalName {
-     *     alias: "valueB"
-     *     fieldC: "valueC"
-     * }
-     *
-     * OverallQueryTransformer will return FieldTransformation for each field in delegate query.
-     * Converting back:
-     * unapply should be called postOrder (for each field in delegate query there is field transformation function that has
-     * some state):
-     * like so:
-     * /orginalName/fieldC: unapply(resultNodeFor(originalName), null) = NamedResultNode(hydration: UnresolvedHydrationField(params))
-     * /orginalName/alias: unapply(resultNodeFor(originalName), null) = NamedResultNode(alias: LeafNode("valueB"))
-     * /orginalName: unapply(resultNodeFor(RootExecutionNode), (two children from above)) = NamedResultNode(fieldA: ObjectExecutionResultNode(children))
-     * </pre>
-     *
-     * @param resultNode the result node
-     * @param children   the result children
-     *
-     * @return a list of named results
-     */
-    default List<NamedResultNode> unapply(ObjectExecutionResultNode resultNode, List<NamedResultNode> children) {
-        //This will be an abstract function
-        throw new UnsupportedOperationException("not ready yet.");
+    default MergedField unapplyMergedField(MergedField mergedField) {
+        return mergedField;
+    }
+
+    default <T extends ExecutionResultNode> T unapplyResultNode(T executionResultNode) {
+        return executionResultNode;
     }
 }
```

Comment:
We now have new unapplyMethods - one for the field AST and one for the result nodes